proto/redpanda/api/dataplane/v1alpha1/error.proto

```diff
@@ -11,4 +11,5 @@ enum Reason {
   REASON_KAFKA_CONNECT_API_ERROR = 5;
   REASON_TYPE_MAPPING_ERROR = 6;
   REASON_SECRET_STORE_ERROR = 7;
+  REASON_FEATURE_NOT_SUPPORTED = 8;
```

Comment:
I think this should go into the common proto package. We already have one similar enum for features: https://buf.build/redpandadata/common/file/main:redpanda/api/common/v1alpha1/common.proto#L41